src/json/value.rs

```diff
@@ -60,8 +60,8 @@ impl<'reg: 'rc, 'rc> From<Json> for ScopedJson<'rc> {
 ///
 #[derive(Debug, Clone)]
 pub struct PathAndJson<'rc> {
```

Comment:
Do we really need these members to be public? There are methods to access them. If it's needed, please add `#[non_exhaustive]` here

Reply:
The problem is that I need string with a `&'rc` lifetime (because that is the key type of BlockParams).
The current accessor functions do not make that possible, and their lifetime annotations cannot be changed to
accomodate that (The helper cannot borrow the whole param for `'rc`).  I guess it would be possible to add a 
`try_get_constant_value(&self) -> &'rc` accessor to that would return `None` for ScopedJson::Derived.
Would you prefer that? Otherwise I'm happy to add `#[non_exhaustive]`.